drivers/sensor/lis2mdl/lis2mdl_trigger.c

```diff
@@ -155,7 +135,18 @@ int lis2mdl_init_interrupt(struct device *dev)
 	lis2mdl->dev = dev;
 #endif
 
-	gpio_pin_enable_callback(lis2mdl->gpio, config->gpio_pin);
+	gpio_pin_configure(lis2mdl->gpio, config->gpio_pin,
```

Comment:
Is this change really related to stmemsc i/f?

Reply:
Not related to stmemsc. I thought it was good to re-write it in the way all the other drivers, at least the ST ones, are done. Moreover, I think that it is better to create the callback thread prior to enable the callback.

Reply:
I would keep it in the same commit even if not strictly related to stmemsc. Do you agree?

Reply:
It's fine to keep in the same PR, but I'd prefer it to be in a separate commit if it's not related to stmemsc

Reply:
ACK.
Done in this way and re-pushed.